src/pages/groupComparison/GenericAssayEnrichments.tsx

```diff
@@ -0,0 +1,130 @@
+import * as React from 'react';
+import { observer } from 'mobx-react';
+import autobind from 'autobind-decorator';
+import { MolecularProfile } from 'cbioportal-ts-api-client';
+import { MakeMobxView } from '../../shared/components/MobxView';
+import Loader from '../../shared/components/loadingIndicator/LoadingIndicator';
+import ErrorMessage from '../../shared/components/ErrorMessage';
+import { MakeEnrichmentsTabUI } from './GroupComparisonUtils';
+import * as _ from 'lodash';
+import ComparisonStore from '../../shared/lib/comparison/ComparisonStore';
+import GenericAssayEnrichmentsContainer from 'pages/resultsView/enrichments/GenericAssayEnrichmentsContainer';
+import EnrichmentsDataSetDropdown from 'pages/resultsView/enrichments/EnrichmentsDataSetDropdown';
+
+export interface IGenericAssayEnrichmentsProps {
+    store: ComparisonStore;
+    genericAssayType: string;
+    resultsViewMode?: boolean;
+}
+
+@observer
+export default class GenericAssayEnrichments extends React.Component<
+    IGenericAssayEnrichmentsProps,
+    {}
+> {
+    @autobind
+    private onChangeProfile(profileMap: {
+        [studyId: string]: MolecularProfile;
+    }) {
+        this.props.store.setGenericAssayEnrichmentProfileMap(
+            profileMap,
+            this.props.genericAssayType
+        );
+    }
+
+    readonly tabUI = MakeEnrichmentsTabUI(
+        () => this.props.store,
+        () => this.enrichmentsUI,
+        this.props.genericAssayType,
+        true,
+        true,
+        false
+    );
+
+    readonly enrichmentsUI = MakeMobxView({
+        await: () => {
+            const ret: any[] = [
+                this.props.store
+                    .genericAssayEnrichmentDataGroupByGenericAssayType,
+                this.props.store
+                    .selectedGenericAssayEnrichmentProfileMapGroupByGenericAssayType,
+                this.props.store
+                    .genericAssayEnrichmentAnalysisGroupsGroupByGenericAssayType,
+                this.props.store.studies,
+            ];
+            if (
+                this.props.store
+                    .genericAssayEnrichmentDataGroupByGenericAssayType
+                    .isComplete &&
+                this.props.store
+                    .genericAssayEnrichmentDataGroupByGenericAssayType.result![
+                    this.props.genericAssayType
+                ]
+            ) {
+                ret.push(
+                    this.props.store
+                        .genericAssayEnrichmentDataGroupByGenericAssayType
+                        .result![this.props.genericAssayType]
+                );
+            }
+            return ret;
+        },
+        render: () => {
+            // since generic assay enrichments tab is enabled only for one study, selectedGenericAssayEnrichmentProfileMap
+            // would contain only one key.
```

Comment:
very nitpicky but in that case the code would be more readable if it was `const studyId = Object.keys(...)[0]` (and leave the comment as well)